### PR TITLE
Validations

### PR DESCRIPTION
**Pull request by NadiaArroyo25**
Many validations are needed in my project, I wanted to add them so that appointments, patients and medical records are not created without the necessary data to identify them

### CODE REVIEW

File: app/models/appointment.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 class Appointment < ApplicationRecord
   validate :delimitation_of_appointment_time
+  validates :title, presence: {message:"es obligatorio"}
+  validates :start_time, presence: {message:"es obligatorio"}
```
Comment (Ligator) [inline app/models/appointment.rb]:
```suggestion
  validates :title, presence: { message: "es obligatorio" }
  validates :start_time, presence: { message: "es obligatorio" }
```

File: app/views/patients/_form.html.erb [hunk 2]
```diff
@@ -1,48 +1,49 @@
...
+        <ul>
+          <% patient.errors.each do |error| %>
+            <li><%= error.full_message %></li>
+          <% end %>
+        </ul>
+      </div>
+    <% end %>
+
+    <div class="form-group">
+      <%= form.label :first_name, style: "display: block"%>
```
Comment (Ligator) [inline app/views/patients/_form.html.erb]: Remove inline styles.
Actually, I don't think you need this style at all.

**Review by Ligator — APPROVED** [review]

File: app/views/appointments/calendar.html.erb [hunk 3]
```diff
@@ -17,8 +17,8 @@
           </div>
           <div class="cars-footer">
             <p class="card-text">
-              <%= appointment.start_time.strftime("%I:%M %p")%>
-              <%= appointment.end_time.strftime("- %I:%M %p")%>
+              <%= appointment.start_time.strftime("%I:%M %p") if appointment.start_time %>
+              <%= appointment.end_time.strftime("- %I:%M %p")if appointment.end_time%>
```
Comment (AlfredoJA) [inline app/views/appointments/calendar.html.erb]:
```suggestion
              <%= appointment.end_time.strftime("- %I:%M %p") if appointment.end_time%>
```

File: app/views/patients/_form.html.erb [hunk 4]
```diff
@@ -1,48 +1,49 @@
...
+      <%= form.text_field :first_name,class: "form-control"%>
```
Comment (AlfredoJA) [inline app/views/patients/_form.html.erb]:
```suggestion
      <%= form.text_field :first_name,class: "form-control" %>
```

File: app/views/patients/_form.html.erb (hunk 2 above)
Comment (AlfredoJA) [inline app/views/patients/_form.html.erb]:
```suggestion
      <%= form.label :first_name, style: "display: block" %>
```

File: app/views/patients/_form.html.erb [hunk 5]
```diff
@@ -1,48 +1,49 @@
...
+    </div>
+
+    <div class="form-group">
+      <%= form.label :address, style: "display: block" %>
+      <%= form.text_field :address, class: "form-control" %>
+    </div>
+
+    <div class="form-group">
+      <%= form.label :age, style: "display: block" %>
+      <%= form.number_field :age , class: "form-control"%>
```
Comment (AlfredoJA) [inline app/views/patients/_form.html.erb]:
```suggestion
      <%= form.number_field :age , class: "form-control" %>
```

**Review by AlfredoJA — APPROVED** [review]

File: app/models/medical_record.rb [hunk 6]
```diff
@@ -1,3 +1,4 @@
 class MedicalRecord < ApplicationRecord
   belongs_to :patient
+  validates :notes, presence:{ message: "es obligatorio" }
```
Comment (GerardoPachecoMesinas) [inline app/models/medical_record.rb]:
```suggestion
  validates :notes, presence: { message: "es obligatorio" }
```

File: app/models/patient.rb [hunk 7]
```diff
@@ -1,3 +1,6 @@
 class Patient < ApplicationRecord
     has_many :medical_records
+    validates :first_name, presence: { message: "es obligatorio" }
+    validates :last_name, presence: { message: " es obligatorio" }
```
Comment (GerardoPachecoMesinas) [inline app/models/patient.rb:4]:
```suggestion
    validates :last_name, presence: { message: "es obligatorio" }
```

File: app/models/patient.rb
Comment (GerardoPachecoMesinas) [inline app/models/patient.rb:1]: one of the three validations does not have the leading space between the quotes, but I don't know which of the two formats you selected

File: app/views/patients/_form.html.erb [hunk 8]
```diff
@@ -1,48 +1,49 @@
...
-  <% if patient.errors.any? %>
-    <div style="color: red">
-      <h2><%= pluralize(patient.errors.count, "error") %> prohibited this patient from being saved:</h2>
-
-      <ul>
-        <% patient.errors.each do |error| %>
-          <li><%= error.full_message %></li>
-        <% end %>
-      </ul>
+<div class= "container ">
```
Comment (GerardoPachecoMesinas) [inline app/views/patients/_form.html.erb]:
```suggestion
<div class="container">
```

File: app/views/patients/_form.html.erb [hunk 9]
```diff
@@ -1,48 +1,49 @@
...
+  <%= form_with(model: patient) do |form| %>
+    <% if patient.errors.any? %>
+      <div style="color: red">
```
Comment (GerardoPachecoMesinas) [inline app/views/patients/_form.html.erb]: don't use inline styles
```suggestion
      <div class="text-danger">
```

File: app/views/appointments/calendar.html.erb [hunk 10]
```diff
@@ -17,8 +17,8 @@
           </div>
           <div class="cars-footer">
             <p class="card-text">
-              <%= appointment.start_time.strftime("%I:%M %p")%>
-              <%= appointment.end_time.strftime("- %I:%M %p")%>
+              <%= appointment.start_time.strftime("%I:%M %p") if appointment.start_time %>
+              <%= appointment.end_time.strftime("- %I:%M %p") if appointment.end_time%>
```
Comment (Ligator) [inline app/views/appointments/calendar.html.erb]:
```suggestion
              <%= appointment.end_time.strftime("- %I:%M %p") if appointment.end_time %>
```

File: app/models/patient.rb [hunk 11]
```diff
@@ -1,3 +1,6 @@
 class Patient < ApplicationRecord
     has_many :medical_records
+    validates :first_name, presence: { message: " es obligatorio" }
+    validates :last_name, presence: { message: " es obligatorio" }
+    validates :age, presence: { message: " es obligatoria" }
```
Comment (Ligator) [inline app/models/patient.rb]: Indentation

File: app/views/patients/_form.html.erb [hunk 12]
```diff
@@ -1,48 +1,49 @@
...
+        <ul>
+          <% patient.errors.each do |error| %>
+            <li><%= error.full_message %></li>
+          <% end %>
+        </ul>
+      </div>
+    <% end %>
+
+    <div class="form-group">
+      <%= form.label :first_name%>
```
Comment (Ligator) [inline app/views/patients/_form.html.erb]:
```suggestion
      <%= form.label :first_name %>
```

Comment (Ligator) [reply app/views/patients/_form.html.erb]: Do a "Find and replace" to fix all these issues.

**Review by Ligator — APPROVED** [review]

**Pull request merged**